src/background_fixed_schedule.rs

```diff
@@ -0,0 +1,399 @@
+}
+
+/// The result of a task to be handled.
+#[derive(Default, Component)]
+pub struct TaskResults<T: TaskWorkerTrait + Send + Sync> {
+    /// The results of the tasks.
+    ///
+    /// This is a queue because we might be spawning a new task while another has not been processed yet.
+    ///
+    /// To avoid overwriting the results, we keep them in a queue.
+    pub results: VecDeque<TaskResult<T>>,
+}
+
+#[derive(Default)]
+pub struct BackgroundFixedUpdatePlugin<T: TaskWorkerTrait> {
+    pub phantom: std::marker::PhantomData<T>,
+}
+
+impl<T: TaskWorkerTrait> Plugin for BackgroundFixedUpdatePlugin<T> {
+    fn build(&self, app: &mut App) {
+        app.add_systems(
+            bevy::app::prelude::RunFixedMainLoop, // TODO: use a specific schedule for this, à la bevy's FixedMainLoop
+            FixedMain::run_schedule::<T>,
+        );
+
+        // this handles checking for task completion, firing writeback schedules and spawning a new task.
+        app.edit_schedule(FixedMain, |schedule| {
+            schedule
+                .add_systems(HandleTask::run_schedule)
+                .set_build_settings(ScheduleBuildSettings {
+                    ambiguity_detection: LogLevel::Error,
+                    ..default()
+                });
+        });
+
+        // those schedules are part of FixedMain
+        app.init_schedule(PreWriteBack);
+        app.edit_schedule(WriteBack, |schedule| {
+            schedule
+                .add_systems(handle_task::<T>)
+                .set_build_settings(ScheduleBuildSettings {
+                    ambiguity_detection: LogLevel::Error,
+                    ..default()
+                });
+        });
+        app.edit_schedule(SpawnTask, |schedule| {
+            schedule
+                .add_systems((extract::<T>, spawn_task::<T>).chain())
+                .set_build_settings(ScheduleBuildSettings {
+                    ambiguity_detection: LogLevel::Error,
+                    ..default()
+                });
+        });
+        app.edit_schedule(PostWriteBack, |schedule| {
+            schedule.set_build_settings(ScheduleBuildSettings {
+                ambiguity_detection: LogLevel::Error,
+                ..default()
+            });
+        });
+    }
+}
+
+/// Difference between tasks and rendering time
+#[derive(Component, Default, Reflect, Clone)]
+pub struct TaskToRenderTime {
+    /// Difference in seconds between tasks and rendering time.
+    ///
+    /// We don't use [`Duration`] because it can be negative.
+    pub diff: f64,
+    /// Amount of rendering frames last task took.
+    pub last_task_frame_count: u32,
+}
+
+/// Difference between tasks and rendering time
+#[derive(Component, Default, Reflect, Clone)]
+pub struct Timestep {
+    pub timestep: Duration,
+}
+
+/// Struct to be able to configure what the task should do.
+/// TODO: extract first, then do work.
+#[derive(Clone, Component)]
+pub struct TaskWorker<T: TaskWorkerTrait> {
+    pub worker: T,
+}
```

Comment:
This should probably be the main component, with required components:
- WorkTask
- Timestep
- TaskToRenderTime
- WorkTask
- TaskResult
- TaskResults